homeassistant/components/solaredge/sensor.py

```diff
@@ -1,102 +1,55 @@
 """Support for SolarEdge Monitoring API."""
-
-from datetime import timedelta
 import logging
-
+import solaredge
 import voluptuous as vol
 
 from requests.exceptions import HTTPError, ConnectTimeout
+
 from homeassistant.components.sensor import PLATFORM_SCHEMA
-from homeassistant.const import (
-    CONF_API_KEY,
-    CONF_MONITORED_CONDITIONS,
-    CONF_NAME,
-    POWER_WATT,
-    ENERGY_WATT_HOUR,
-)
+from homeassistant.config_entries import SOURCE_IMPORT
+from homeassistant.const import CONF_API_KEY, CONF_NAME
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entity import Entity
 from homeassistant.util import Throttle
 
-# Config for solaredge monitoring api requests.
-CONF_SITE_ID = "site_id"
-
-OVERVIEW_UPDATE_DELAY = timedelta(minutes=10)
-DETAILS_UPDATE_DELAY = timedelta(hours=12)
-INVENTORY_UPDATE_DELAY = timedelta(hours=12)
-POWER_FLOW_UPDATE_DELAY = timedelta(minutes=10)
-
-SCAN_INTERVAL = timedelta(minutes=10)
-
-# Supported overview sensor types:
-# Key: ['json_key', 'name', unit, icon]
-SENSOR_TYPES = {
-    "lifetime_energy": [
-        "lifeTimeData",
-        "Lifetime energy",
-        ENERGY_WATT_HOUR,
-        "mdi:solar-power",
-    ],
-    "energy_this_year": [
-        "lastYearData",
-        "Energy this year",
-        ENERGY_WATT_HOUR,
-        "mdi:solar-power",
-    ],
-    "energy_this_month": [
-        "lastMonthData",
-        "Energy this month",
-        ENERGY_WATT_HOUR,
-        "mdi:solar-power",
-    ],
-    "energy_today": [
-        "lastDayData",
-        "Energy today",
-        ENERGY_WATT_HOUR,
-        "mdi:solar-power",
-    ],
-    "current_power": ["currentPower", "Current Power", POWER_WATT, "mdi:solar-power"],
-    "site_details": [None, "Site details", None, None],
-    "meters": ["meters", "Meters", None, None],
-    "sensors": ["sensors", "Sensors", None, None],
-    "gateways": ["gateways", "Gateways", None, None],
-    "batteries": ["batteries", "Batteries", None, None],
-    "inverters": ["inverters", "Inverters", None, None],
-    "power_consumption": ["LOAD", "Power Consumption", None, "mdi:flash"],
-    "solar_power": ["PV", "Solar Power", None, "mdi:solar-power"],
-    "grid_power": ["GRID", "Grid Power", None, "mdi:power-plug"],
-    "storage_power": ["STORAGE", "Storage Power", None, "mdi:car-battery"],
-}
+from .const import (
+    DOMAIN,
+    CONF_SITE_ID,
+    OVERVIEW_UPDATE_DELAY,
+    DETAILS_UPDATE_DELAY,
+    INVENTORY_UPDATE_DELAY,
+    POWER_FLOW_UPDATE_DELAY,
+    SENSOR_TYPES,
+)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
Are we sure that there will never be more platforms than the sensor platform? If there's any chance of more platforms we should move the config section and schema to the component.

Reply:
I just tried to keep the component as close as possible to the original one and just add the config flow.

Until know the solaredge api does not support anything more then sensors, i don't think there will be other platforms.

Reply:
What about some kind of control of solar panel equipment or just binary sensors?

Reply:
until now its not there,
if we move the config to the __init__ then the solaredge config in configuration.yaml is not working anymore

Reply:
We would move the config section to the solaredge key in configuration.yaml.

Reply:
True willl move them

Reply:
maybe we can even remove the config from configuration.yaml.
why would we keep it there if we support config entries

Reply:
Some users like using configuration.yaml to keep their config version controlled etc.